src/main/java/com/user_service/services/UserAuthenticationService.java

```diff
@@ -45,23 +52,24 @@ public SignUpResponseDto signUpUser(RegisterUserDto registerUserDto) {
         try {
             SignUpRequest signUpRequest = SignUpRequest.builder()
                     .userAttributes(attrs)
-                    .username(registerUserDto.getEmail())
+                    .username(signUpUserDto.getEmail())
                     .clientId(CLIENT_ID)
-                    .password(registerUserDto.getPassword())
+                    .password(signUpUserDto.getPassword())
                     .build();
 
+            log.info("New user with email: {} created", signUpUserDto.getEmail());
             return new SignUpResponseDto(cognitoIdentityProviderClient.signUp(signUpRequest), null);
-
-        } catch(CognitoIdentityProviderException e) {
+        } catch (CognitoIdentityProviderException e) {
+            log.error("Failed to sign up user, error: {}", e.awsErrorDetails().errorMessage());
             return new SignUpResponseDto(null, e.awsErrorDetails().errorMessage());
         }
     }
 
-    public void signInUser(LoginUserDto loginUserDto) {
+    public SignInResponseDto signInUser(SignInUserDto signInUserDto) {
         try {
-            Map<String, String> authParams = new LinkedHashMap<String, String>() {{
-                put("USERNAME", loginUserDto.getEmail());
-                put("PASSWORD", loginUserDto.getPassword());
+            Map<String, String> authParams = new LinkedHashMap<>() {{
```

Comment:
A potential code readability improvement would be to hide the Cognito client inside a wrapper class like e.g. `ExternalCognitoClient` to hide its not very pleasant to use interface. It's a bit like [Anticorruption Layer](https://www.oreilly.com/library/view/learning-domain-driven-design/9781098100124/ch04.html) - though not exaclty.